cpp/daal/src/services/env_detect.cpp

```diff
@@ -125,8 +125,12 @@ DAAL_EXPORT void daal::services::Environment::setDynamicLibraryThreadingTypeOnWi
     initNumberOfThreads();
 }
 
-DAAL_EXPORT daal::services::Environment::Environment() : _globalControl {}
+DAAL_EXPORT daal::services::Environment::Environment() : _schedulerHandle {}, _globalControl {}
 {
+    // When the oneapi::tbb::finalize function is called with an oneapi::tbb::task_scheduler_handle
+    // instance, it blocks the calling thread until the completion of all worker
+    // threads that were implicitly created by the library.
+    daal::setSchedulerHandle(&_schedulerHandle);
```

Comment:
```suggestion
    // Initializes global oneapi::tbb::task_scheduler_handle object in oneDAL to prevent the unexpected
    // destruction of the calling thread.
    // When the oneapi::tbb::finalize function is called with an oneapi::tbb::task_scheduler_handle
    // instance, it blocks the calling thread until the completion of all worker
    // threads that were implicitly created by the library.
    daal::setSchedulerHandle(&_schedulerHandle);
```